src/api/search/src/search.js

```diff
@@ -103,4 +101,57 @@ const search = async (options) => {
     values: hits.hits.map(({ _id }) => ({ id: _id, url: `${POSTS_URL}/${_id}` })),
   };
 };
-module.exports = { search };
+
+/**
+ * authorAutocomplete allows for querying author autocomplete fields
+ * @param { author } - text to search in author field
+ * @return all the results matching the fields text
+ */
+const authorAutocomplete = async ({ author }) => {
+  const results = {
+    query: {
+      match: {
+        'author.autocomplete': {
+          query: author,
+          operator: 'and',
+        },
+      },
+    },
+    highlight: {
+      fields: {
+        'author.autocomplete': {},
+      },
+    },
+  };
+
+  const {
+    body: { hits },
+  } = await client.search({
+    size: 10000,
+    _source: ['author'],
+    index,
+    body: results,
+  });
+
+  // Filter through all authors and remove duplicates
+  const authors = hits.hits.reduce(
+    (acc, { _source }) => (acc.includes(_source.author) ? acc : [...acc, _source.author]),
```

Comment:
```suggestion
    (acc, { _source }) => (acc.includes(_source.author) ? acc : acc.concat(_source.author),
```

Reply:
I'm just curious if there's a reason to use `concat` over what was before. Better readability?

Reply:
Readability yeah, why use spread syntax when you can use a built-in function. 
I would personal write `acc.push(_source.auth)` to get O(1) 🤓

Reply:
Except I think `acc.push` just returns the new length of the array. To use it we can't use it in one line and would still have to `return acc` somewhere in the callback, overall O(n). 

I guess they're all the same. I'll change to `concat` for better readability.

Reply:
Correct me if im wrong but arrays are JS objects and are passed by reference. So if you push to `acc` and return it, it's not copied(O(n)) but reused in the next iteration.

Reply:
JS object is passed by reference

Reply:
None of this matters, we're talking about 10, 100, 1000 iterations?  That's not where this code is going to be slow.  What's slow is the network transfer.  Don't optimize the wrong thing.

Reply:
I agree, it's premature optimization , won't make much difference. We just discussing for the best solution